kbrnsr-playground/backend-stuff-elixir-basics
# Move all Mix projects into umbrella project (#9)

* add umbrella mix project executed "mix new elixir_basics --umbrella" in repo root
* rename numbers to elixir_basics/apps/numbers
renamed:    numbers -> elixir_basics/apps/numbers
renamed:    numbers/.formatter.exs -> elixir_basics/apps/numbers/.formatter.exs
renamed:    numbers/.gitignore -> elixir_basics/apps/numbers/.gitignore
renamed:    numbers/README.md -> elixir_basics/apps/numbers/README.md
renamed:    numbers/lib/basic_math.ex -> elixir_basics/apps/numbers/lib/basic_math.ex
renamed:    numbers/lib/calculator.ex -> elixir_basics/apps/numbers/lib/calculator.ex
renamed:    numbers/mix.exs -> elixir_basics/apps/numbers/mix.exs
renamed:    numbers/test/basic_math_test.exs -> elixir_basics/apps/numbers/test/basic_math_test.exs
renamed:    numbers/test/calculator_test.exs -> elixir_basics/apps/numbers/test/calculator_test.exs
renamed:    numbers/test/test_helper.exs -> elixir_basics/apps/numbers/test/test_helper.exs
* changed elixir_basics/apps/numbers config
Changed so it can work inside the Mix umbrella project
* rename elixir_basics/apps/lib/* files to numbers/*
I don't want to do it this way, but iex prepends the modules with
"Elixir." if I refuse to do this, but at least I don't have to do any additional configuration.
renamed:    elixir_basics/apps/numbers/lib/basic_math.ex -> elixir_basics/apps/numbers/lib/numbers/basic_math.ex
renamed:    elixir_basics/apps/numbers/lib/calculator.ex -> elixir_basics/apps/numbers/lib/numbers/calculator.ex
* rename recursion to elixir_basics/apps/recursion
renamed:    recursion -> elixir_basics/apps/recursion
renamed:    recursion/.formatter.exs -> elixir_basics/apps/recursion/.formatter.exs
renamed:    recursion/.gitignore -> elixir_basics/apps/recursion/.gitignore
renamed:    recursion/README.md -> elixir_basics/apps/recursion/README.md
renamed:    recursion/lib/recursion.ex -> elixir_basics/apps/recursion/lib/recursion.ex
renamed:    recursion/mix.exs -> elixir_basics/apps/recursion/mix.exs
renamed:    recursion/test/recursion_partial_occurence_test.exs -> elixir_basics/apps/recursion/test/recursion_partial_occurence_test.exs
renamed:    recursion/test/test_helper.exs -> elixir_basics/apps/recursion/test/test_helper.exs
* change elixir_basics/apps/recursion mix config
* rename linked_lists
renamed:    linked_lists -> elixir_basics/apps/linked_lists
renamed:    linked_lists/.formatter.exs -> elixir_basics/apps/linked_lists/linked_lists/.formatter.exs
renamed:    linked_lists/.gitignore -> elixir_basics/apps/linked_lists/linked_lists/.gitignore
renamed:    linked_lists/README.md -> elixir_basics/apps/linked_lists/linked_lists/README.md
renamed:    linked_lists/lib/linked_lists.ex -> elixir_basics/apps/linked_lists/linked_lists/lib/linked_lists.ex
renamed:    linked_lists/mix.exs -> elixir_basics/apps/linked_lists/linked_lists/mix.exs
renamed:    linked_lists/test/linked_lists_test.exs -> elixir_basics/apps/linked_lists/linked_lists/test/linked_lists_test.exs
renamed:    linked_lists/test/test_helper.exs -> elixir_basics/apps/linked_lists/linked_lists/test/test_helper.exs
* rename apps/linked_lists/linked_lists
renamed:    elixir_basics/apps/linked_lists/linked_lists -> elixir_basics/apps/linked_lists
renamed:    elixir_basics/apps/linked_lists/linked_lists/README.md -> elixir_basics/apps/linked_lists/README.md
renamed:    elixir_basics/apps/linked_lists/linked_lists/lib/linked_lists.ex -> elixir_basics/apps/linked_lists/lib/linked_lists.ex
renamed:    elixir_basics/apps/linked_lists/linked_lists/mix.exs -> elixir_basics/apps/linked_lists/mix.exs
renamed:    elixir_basics/apps/linked_lists/linked_lists/test/linked_lists_test.exs -> elixir_basics/apps/linked_lists/test/linked_lists_test.exs
renamed:    elixir_basics/apps/linked_lists/linked_lists/test/test_helper.exs -> elixir_basics/apps/linked_lists/test/test_helper.exs
* update apps/linked_lists mix config
* fix failing test in apps/linked_lists/test
* rename blork to elixir_basics/apps/blork
renamed:    blork -> elixir_basics/apps/blork
renamed:    blork/.formatter.exs -> elixir_basics/apps/blork/.formatter.exs
renamed:    blork/.gitignore -> elixir_basics/apps/blork/.gitignore
renamed:    blork/README.md -> elixir_basics/apps/blork/README.md
renamed:    blork/lib/blork.ex -> elixir_basics/apps/blork/lib/blork.ex
renamed:    blork/mix.exs -> elixir_basics/apps/blork/mix.exs
renamed:    blork/mix.lock -> elixir_basics/apps/blork/mix.lock
renamed:    blork/test/blork_test.exs -> elixir_basics/apps/blork/test/blork_test.exs
renamed:    blork/test/test_helper.exs -> elixir_basics/apps/blork/test/test_helper.exs
* update blork mix config
* rename elixir_basics/apps/blork/mix.lock
renamed:    elixir_basics/apps/blork/mix.lock -> elixir_basics/mix.lock
* update blork.ex to suppress compiler warning
I was getting "warning: function private_hello/0 is unused" while
compiling the umbrella project. I decided to just call it from the
Blork.hello/0 function to suppress it.
* rename all_strings mix project
renamed:    all_strings -> elixir_basics/apps/all_strings
renamed:    all_strings/.formatter.exs -> elixir_basics/apps/all_strings/.formatter.exs
renamed:    all_strings/.gitignore -> elixir_basics/apps/all_strings/.gitignore
renamed:    all_strings/README.md -> elixir_basics/apps/all_strings/README.md
renamed:    all_strings/lib/all_strings.ex -> elixir_basics/apps/all_strings/lib/all_strings.ex
renamed:    all_strings/mix.exs -> elixir_basics/apps/all_strings/mix.exs
renamed:    all_strings/test/all_strings_test.exs -> elixir_basics/apps/all_strings/test/all_strings_test.exs
renamed:    all_strings/test/test_helper.exs -> elixir_basics/apps/all_strings/test/test_helper.exs
* update elixir_basics/apps/all_strings mix config
* fix all_strings/test failure
The test file still had the hello test in it, I just added a test for
trimming whitespace around a string
* update README
Updated README with references to new folder structure as well as a few corrections.
* update elixirls config
I removed the workspace file since it isn't needed anymore with the
Mix umbrella project. I also added some config so ElixirLS understands that the project root should be elixir_basics and not the repo root.
